src/Security/SecureSigner.php

```diff
@@ -47,16 +51,47 @@ public function createSignature($method, $uri, array $nameValuePairs, $body = ""
     }
 
     /**
+     * @param array $params either headers or request params in format [ key => value ]
+     * @throws Exception
+     */
+    public function validateFormRedirect(array $params )
+    {
+        $this->validateSignature("GET", "", $params);
+    }
+
+    /**
+     * @param $method
+     * @param $uri
+     * @param array $nameValuePairs either headers or request params in format [ key => value ]
+     * @param string $body
+     * @throws Exception
+     */
+    public function validateSignature($method, $uri, array $nameValuePairs, $body = "")
+    {
+
+        $expectedSignature = $nameValuePairs["signature"];
```

Comment:
I think expected and actual signatures should be swapped, as we expect the one we calculate and actual was the one given :) In java lib the term "receivedSignature" is used